source/storage/apis-1.5.rst

```diff
@@ -432,6 +509,47 @@ Request Headers
     response will be returned.
 
 
+**X-Weave-Records**
+
+    This header may be sent with multi-record uploads, to indicate the
+    total number of records included in the request.  If the server
+    would not accept an upload containing that many records, then a
+    **400 Bad Request** response will be returned with response code **17**.
+
+
+**X-Weave-Bytes**
+
+    This header may be sent with multi-record uploads, to indicate the
+    combined size of payloads in the upload, in bytes.  If the server
+    would not accept an upload containing that many bytes, then a
+    **400 Bad Request** response will be returned with response code **17**.
+
+
+**X-Weave-Batch-Records**
+
+    This header may be included with a POST request using the **batch** query
+    parameter, to indicate the total number of records in the batch.
+    If the server would not accept a batch containing that many records,
+    then a **400 Bad Request** response will be returned with response
+    code **17**.
+
+    If the value of this header is not a valid positive integer value, or if
+    the request is not operating on a batch, then a **400 Bad Request**
+    response will be returned with response code **1**.
+
+**X-Weave-Batch-Bytes**
+
+    This header may be included with a POST request using the **batch** query
+    parameter, to indicate the total combined size of payloads in the batch,
```

Comment:
Ah, combined. Perhaps let's use X-Weave-Batch-Whatever for the individual batches, and X-Weave-Total-Whatever for the combination?